exporter/collector/integrationtest/testdata/fixtures/metrics/google_managed_prometheus_expect.json

```diff
@@ -942,7 +942,6 @@
               "http_scheme": "http",
               "k8s_container_name": "rabbitmq",
               "k8s_node_name": "10.92.5.2",
-              "k8s_pod_name": "rabbitmq-server-0",
```

Comment:
Do you know why this gets removed?

Reply:
yup now that `k8s.pod.name` gets parsed to `job` the extra metrics function ignores it: https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/2be0414d564ce62759e2da63615b27103ab950d3/exporter/collector/googlemanagedprometheus/extra_metrics.go#L180-L185

Reply:
Thats unfortunate, since we may have only added the service.name to the monitored resource.  It would be nice if we only dropped labels that were _actually_ used in the monitored resource, but i'm not sure how hard that is to do.

Reply:
No you're totally right, I assumed this was still being preserved but I see it's not. This shouldn't be dropped, I'll have to fix that logic

Reply:
Updated in https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/823/commits/1fbb3d532461dfcf4085e48900a2e44e8a92ead3 based on offline discussion